src/btree/bt_split.c

```diff
@@ -1786,15 +1827,19 @@ __split_insert(WT_SESSION_IMPL *session, WT_REF *ref)
 	    page_decr, right_incr, __wt_update_list_memsize(moved_ins->upd));
 
 	/*
-	 * Allocation operations completed, move the last insert list item from
-	 * the original page to the new page.
+	 * Move the last insert list item from the original page to the new
+	 * page.
 	 *
 	 * First, update the item to the new child page. (Just append the entry
 	 * for simplicity, the previous skip list pointers originally allocated
 	 * can be ignored.)
 	 */
-	right->pg_row_ins[0]->head[0] =
-	    right->pg_row_ins[0]->tail[0] = moved_ins;
+	if (type == WT_PAGE_ROW_LEAF)
+		right->pg_row_ins[0]->head[0] =
+		    right->pg_row_ins[0]->tail[0] = moved_ins;
+	else
+		right->modify->mod_append[0]->head[0] =
+		    right->modify->mod_append[0]->tail[0] = moved_ins;
```

Comment:
This makes my head hurt.  Can we set a `WT_INSERT_HEAD` local variable, so this becomes something like:

```c
inshead = (type == WT_PAGE_ROW_LEAF) ?
    right->pg_row_ins[0] : right->modify->mod_append[0];
inshead->head[0] = inshead->tail[0] = moved_ins;
```